protocol/x/clob/ante/clob.go

```diff
@@ -250,15 +256,93 @@ func IsShortTermClobMsgTx(ctx sdk.Context, tx sdk.Tx) (bool, error) {
 
 	numMsgs := len(msgs)
 	if numMsgs > 1 {
-		return false, errorsmod.Wrap(
+		return true, errorsmod.Wrap(
 			sdkerrors.ErrInvalidRequest,
-			"a transaction containing MsgCancelOrder or MsgPlaceOrder may not contain more than one message",
+			"a transaction containing short term MsgCancelOrder or MsgPlaceOrder may not contain more than one message",
 		)
 	}
 
 	return true, nil
 }
 
+// HasClobMsg returns `true` if the transaction has at least one clob msg
+func HasClobMsg(tx sdk.Tx) bool {
+	msgs := tx.GetMsgs()
+
+	for _, msg := range msgs {
+		switch msg.(type) {
+		case *types.MsgCancelOrder:
+			return true
+		case *types.MsgPlaceOrder:
+			return true
+		case *types.MsgBatchCancel:
+			return true
+		}
+	}
+	return false
+}
+
+// IsValidClobMsgTx checks if the transaction contains a valid set of clob msgs
+// This function assumes that the input tx has at least one clob msg
+// A transaction with a clob msg must adhere to the below conditions
+//   - If the tx contains a short term order msg, the tx can only have one msg
+//   - If the tx contains a stateful order msg, it can only contain other stateful order msgs
+//     or a single transfer msg
+func IsValidClobMsgTx(tx sdk.Tx) error {
+	msgs := tx.GetMsgs()
+
+	var hasShortTermOrder = false
+	var numTransferMsgs = 0
+	// Non CLOB msgs are not allowed in CLOB msg transactions because there is no fee charged
+	// for CLOB transactions
+	var hasNonClobMsg = false
```

Comment:
Nit: naming is not accurate since `MsgCreateTransfer` is non-clob. `hasDisallowedMsg`?